server/src/main/java/org/elasticsearch/common/xcontent/XContentElasticsearchExtension.java

```diff
@@ -103,10 +103,8 @@ public Map<Class<?>, XContentBuilder.HumanReadableTransformer> getXContentHumanR
     public Map<Class<?>, Function<Object, Object>> getDateTransformers() {
         Map<Class<?>, Function<Object, Object>> transformers = new HashMap<>();
         transformers.put(Date.class, d -> DEFAULT_FORMATTER.format(((Date) d).toInstant()));
-        transformers.put(Long.class, d -> DEFAULT_FORMATTER.format(Instant.ofEpochMilli((long) d)));
         transformers.put(Calendar.class, d -> DEFAULT_FORMATTER.format(((Calendar) d).toInstant()));
         transformers.put(GregorianCalendar.class, d -> DEFAULT_FORMATTER.format(((Calendar) d).toInstant()));
-        transformers.put(Instant.class, d -> DEFAULT_FORMATTER.format((Instant) d));
```

Comment:
Why the removal of Instant? I understand not wanting a primitive type like long to implicitly be a date, but Instant is a datetime class?

Reply:
`Instant` was in the map twice, this entry was overwritten by the other one.